feature/google/src/unityIAP/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -704,7 +715,53 @@ class BillingWrapper(
                 }
                 hasResponded = true
             }
+            trackGoogleQueryPurchaseHistoryRequestIfNeeded(billingResult, requestStartTime)
             listener.onPurchaseHistoryResponse(billingResult, purchaseHistory)
         }
     }
+
+    private fun BillingClient.queryPurchasesAsyncWithTracking(
+        @BillingClient.SkuType skuType: String,
+        listener: PurchasesResponseListener
+    ) {
+        val requestStartTime = dateProvider.now.time
+        queryPurchasesAsync(skuType) { billingResult, purchases ->
+            trackGoogleQueryPurchasesRequestIfNeeded(skuType, billingResult, requestStartTime)
+            listener.onQueryPurchasesResponse(billingResult, purchases)
+        }
+    }
+
+    private fun trackGoogleQuerySkuDetailsRequestIfNeeded(
+        @BillingClient.SkuType skuType: String,
+        billingResult: BillingResult,
+        requestStartTime: Long
+    ) {
+        diagnosticsTracker?.trackGoogleQuerySkuDetailsRequest(
+            skuType,
+            billingResult.responseCode,
+            billingResult.debugMessage,
+            responseTimeMillis = dateProvider.now.time - requestStartTime
+        )
+    }
+
+    private fun trackGoogleQueryPurchasesRequestIfNeeded(
+        @BillingClient.SkuType skuType: String,
+        billingResult: BillingResult,
+        requestStartTime: Long
+    ) {
+        diagnosticsTracker?.trackGoogleQueryPurchasesRequest(
+            skuType,
+            billingResult.responseCode,
+            billingResult.debugMessage,
+            responseTimeMillis = dateProvider.now.time - requestStartTime
+        )
+    }
+
+    private fun trackGoogleQueryPurchaseHistoryRequestIfNeeded(billingResult: BillingResult, requestStartTime: Long) {
+        diagnosticsTracker?.trackGoogleQueryPurchaseHistoryRequest(
```

Comment:
Currently we understand that diagnostics is disabled if this class is null. I think that's ok, but I'm not sure if it's explicit enough... Not planning to change it for now but lmk if you have any concerns.